userspace/libsinsp/sinsp.cpp

```diff
@@ -1426,25 +1423,20 @@ int32_t sinsp::next(OUT sinsp_evt **puevt)
 	// things like exit() or close() can be parsed.
 	//
 	uint32_t nfdr = (uint32_t)m_fds_to_remove->size();
-
 	if(nfdr != 0)
 	{
 		/* This is a removal logic we shouldn't scan /proc. If we don't have the thread 
 		 * to remove we are fine.
 		 */
 		sinsp_threadinfo* ptinfo = get_thread_ref(m_tid_of_fd_to_remove, false).get();
-		if(!ptinfo)
-		{
-			ASSERT(false);
-			return res;
-		}
-
-		for(uint32_t j = 0; j < nfdr; j++)
+		if(ptinfo)
 		{
-			ptinfo->remove_fd(m_fds_to_remove->at(j));
+			for(uint32_t j = 0; j < nfdr; j++)
+			{
+				ptinfo->remove_fd(m_fds_to_remove->at(j));
+			}
+			m_fds_to_remove->clear();
```

Comment:
Moving the ->clear() into the if{} changes the behavior: now we'll keep trying to remove the fds until we succeed, and we might succeed when the tinfo points to a completely different process with the same pid.

OTOH, if the tinfo is not there in the thread table, its fds are presumably also gone. So I'd say move the ->clear() outside.

Reply:
I am not sure: if we hadn't got `ptinfo` we always returned, never calling `m_fds_to_remove->clear();`; this is  the same as now, right?
But i guess you are right, it should make sense to call it in any case.

cc @Andreagit97 : does it make sense to call `m_fds_to_remove->clear();` in any case, even if `ptinfo` is NULL?

Reply:
ah good point. -ENOBRAIN. I think I'd still like the change but as a separate commit (so we don't wonder about this in a few months)

Reply:
Pushed the change (coauthored :P)
Thank you!